### ISSUE TITLE
Cannot get Android application from GooglePlay.

### ISSUE THREAD

**Issue by sebsauvage** [keybase/client#6954]
Phone: Samsung Galaxy S5 new, Android 6.0.1

From GooglePlay (on the phone): Google says the application is not available in my country. (I'm in France).
(Screenshot: https://framapiaf.org/system/media_attachments/files/000/045/471/original/a8a53eef29f73e10.png )

From GooglePlay website, try to install: Google says the application is not compatible with my phone (???).
(Screenshot: https://framapiaf.org/system/media_attachments/files/000/045/488/original/f32eb0a29d4fcf2b.png )

Those two look like weak excuses.
Could you just provide the APK on your website ?

**Comment by cjb** [comment]
Hi!  Unfortunately France is the only country the app isn't live in.  It's because France requires permission from the government before importing crypto from other countries:

https://en.wikipedia.org/wiki/Cryptography_law#France

We hope it will be in the store in the future.

**Comment by sebsauvage** [comment]
But the desktop applications are freely downloadable from your website.
Will you also pose a restriction ?

Could you provide a direct apk download link on your website ?  Or a least an md5 of the apk ?
I would not want to get Keybase from dubious third-party stores.

**Comment by sebsauvage** [comment]
I don't really get it: Signal, WhatsApp and Telegram use strong crypto too, and they are available on GooglePlay.

**Comment by renaudManda** [comment]
@sebsauvage did find a solution about this issue ?

**Comment by sebsauvage** [comment]
In order to try the app, I downloaded it from a third party website. :-(

**Comment by shoxxdj** [comment]
As our administration is not the faster on this planet, and download from a third party website is not in my religion, can you provide a apk here ? :D 
Thanks for French people !

**Comment by renaudManda** [comment]
@sebsauvage I did it too. @cjb I don't think that publishing an hash will be outlaw :)

**Comment by sebsauvage** [comment]
> I don't think that publishing an hash will be outlaw :)

Indeed !

**Comment by treyssatvincent** [comment]
> Exportation and importation of cryptographic tools to or from foreign countries
Maybe a fork from a French can bypass this liberticidal law.

**Comment by np** [comment]
Can these hashes be confirmed ?

```
sha256sum io.keybase.ossifrage_1.0.2-724749_minAPI16(armeabi-v7a,x86)(nodpi)_apkmirror.com.apk
a7996f55eef2a7dd7cf87e8ac82ea17cfacbccee07cef0ae6c14685cb7daaa27
```

**Comment by Dryusdan** [comment]
@renaudManda Hello
I'm in France to and can't download app to (stupid useless law...)
But... It's possible to put apk in the release section of Github ? :) Like that you not publish app in France ;)

**Comment by stormtrooper42** [comment]
To bypass the law just distribute the apk here or on f-droid ;)

**Comment by hferee** [comment]
My understanding is that it may only require declaration, not authorisation.
If it's not possible, you might as well want to change your moto to "Crypto for (almost) everyone!"

**Comment by treyssatvincent** [comment]
@hferee `requires an explicit authorization (for countries outside the EU).`

It's France who should change of moto to : "(Almost) Liberté, Egalité Fraternité".

**Comment by hferee** [comment]
@treyssatvincent Indeed, wikipedia is not a source.

Let's get accurate if you like.
The text you refer to states (Section 1, III) states that "importing cryptology tools are required to be declared to the Prime Minister".

Here, the Prime Minister is represented by our IS security agency (ANSSI), whose web page clearly says "declaration": 
https://www.ssi.gouv.fr/administration/reglementation/controle-reglementaire-sur-la-cryptographie/demarches-a-accomplir/

The forms are on the same website. I'm not saying it's not a hassle, but don't make it look worse than it is.

**Comment by stormtrooper42** [comment]
Just publish you app !  Protonmail is outside the EU and they didn't make any declaration in france.

**Comment by SamouraiDev** [comment]
As far as downloading official APK via Play in France, try this: create throwaway US-based Google account, empty Play cache and download. Worked for me. For good measure, do connection over VPN.

**Comment by ZerBea** [comment]
Samsung Galaxy i9100 Lineage 14.1
Could be a nice idea to provide your APK outside the "world of google" - direct download or via F-Droid.

**Comment by Depado** [comment]
I agree that it should be possible to download the APK from GH releases or directly from your website. There are also a number of people not wanting to use the Google ecosystem to download apps. 
Thanks a lot for your great work :)

**Comment by mikedilger** [comment]
I'm not in France, but I refuse to let Google into my phone.  So I also would need a trusted source for the APK, such as F-Droid, downloading from keybase.io, or signed and put pretty much anywhere.

**Comment by HorlogeSkynet** [comment]
Salut @sebsauvage @Dryusdan, je connais bien ces pseudos, à force de les lire sur Mastodon par l'intermédiaire d'Angristan 🤣

About the issue : What the **** ? There are many applications using point-to-point encryption, but you can't publish Keybase on the Play Store ?! This is insane 😂

And yeah, of course, 👍 for F-Droid, it'll help many many many many (many) people using custom ROMs, without GApps.

A big thanks anyway, keep on 💪

**Comment by solvaholic** [comment]
Per https://github.com/keybase/client/issues/6899 this issue [has been resolved](https://twitter.com/malgorithms/status/915197913860657152).

**Comment by MisterY** [comment]
After reading 6899, the issue is nowhere near being resolved.

**Comment by ZerBea** [comment]
For me, it is. Now using SIGNAL. Direct download from their web site:
https://signal.org/android/apk/
